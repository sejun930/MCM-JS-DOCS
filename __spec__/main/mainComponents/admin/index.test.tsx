import { render, act } from "@testing-library/react";
import { RecoilRoot } from "recoil";

import AdminMainHome from "src/main/mainComponents/admin";

let container: HTMLElement;
// 관리자 메인 홈페이지
describe("Admin Main Home Page", () => {
  // 비동기 함수 처리하는 mock 함수
  jest.mock("../../../../src/main/commonsComponents/withAuth", () => ({
    __esModule: true,
    default: jest.fn(() => Promise.resolve(true)), // Resolve with a mock value
  }));

  // 비동기 함수 처리 및 container 초기화
  beforeEach(async () => {
    await act(async () => {
      container = (
        await render(
          <RecoilRoot>
            <AdminMainHome />
          </RecoilRoot>
        )
      ).container;
    });
  });

  // 스냅샷
  test("Admin Main Home Page - Snapshot", async () => {
    expect(container).toMatchSnapshot();
  });

  // withAuth 체크
  test("Admin Main Home Page - Check ID Selector", () => {
    const checkId = container.querySelector("#main-template-wrapper");
    expect(checkId).toBeInTheDocument();
  });
});
