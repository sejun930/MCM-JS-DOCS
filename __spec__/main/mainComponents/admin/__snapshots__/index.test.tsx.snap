// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Main Home Page Admin Main Home Page - Snapshot 1`] = `
<div>
  <main
    class="project-remarks-template admin-withAuth-template css-1hqyx7v"
    id="main-template-wrapper"
  >
    <nav
      class="css-ggsnke"
    >
      <button
        class="mcm-button-unit css-y7od8r"
        role="button_click_event"
        type="submit"
      >
        üè†
      </button>
      <button
        class="mcm-button-unit mobile-nav-button css-zfv12t"
        role="button_click_event"
        type="submit"
      />
    </nav>
    <section
      class="css-1bsxzru"
    >
      <div
        class="css-1d8bsng"
      >
        <div
          class="mcm-modal-wrapper admin-login-modal modal-use-fixed-window css-ybxyti"
        >
          <div
            class="mcm-modal-items css-1fs3plo"
          >
            <div
              class="mcm-modal-close-button-wrapper css-me5gxd"
            >
              <button
                class="mcm-button-unit mcm-modal-close-ment-button css-19ngahp"
                role="button_click_event"
                type="button"
              >
                <span
                  class="mcm-span-unit mcm-modal-close-ment css-1oteowz"
                >
                  ÌôàÏúºÎ°ú Ïù¥Îèô
                </span>
              </button>
              <button
                class="mcm-close-button-unit mcm-modal-close-button css-14jhqbo"
                style="display: flex;"
                type="button"
              />
            </div>
            <div
              class="mcm-modal-contents css-1c4r0bf"
            >
              <div
                class="css-1sqmh5d"
                id="admin-login-wrapper"
              >
                <h1
                  class="mcm-title-unit"
                  id="admin-login-title"
                >
                  <p>
                    ÎπÑÎ°úÍ∑∏Ïù∏ ÏÉÅÌÉúÏûÖÎãàÎã§.
                  </p>
                  <p>
                    Í¥ÄÎ¶¨ÏûêÎ°ú 
                    <b>
                      Î°úÍ∑∏Ïù∏
                    </b>
                    Ìï¥Ï£ºÏÑ∏Ïöî.
                  </p>
                </h1>
                <form
                  class="css-evrzgm"
                >
                  <div
                    class="css-bfo5ge"
                  >
                    <div
                      class="css-j3wzer"
                    >
                      <div
                        class="mcm-input-unit-wrapper css-ey5vfm"
                        id="admin-id-input"
                      >
                        <fieldset
                          class="css-1tsvgnq"
                        >
                          <legend>
                            Input & TextArea
                          </legend>
                        </fieldset>
                        <div
                          class="mcm-input-unit-items css-1bcrtr8"
                        >
                          <input
                            autocomplete="off"
                            class="mcm-input-unit css-1348qyx"
                            maxlength="30"
                            placeholder="Í¥ÄÎ¶¨Ïûê ÏïÑÏù¥ÎîîÎ•º ÏûÖÎ†•Ìï¥Ï£ºÏÑ∏Ïöî."
                            type="text"
                            value=""
                          />
                          <div
                            class="mcm-input-submit-button-wrapper css-cakho0"
                          >
                            <div
                              class="mcm-input-submit-button-items css-16nhp7q"
                            >
                              <button
                                class="mcm-close-button-unit css-12tnlcr"
                                type="button"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="css-j3wzer"
                    >
                      <div
                        class="mcm-input-unit-wrapper css-ey5vfm"
                        id="admin-pw-input"
                      >
                        <fieldset
                          class="css-1tsvgnq"
                        >
                          <legend>
                            Input & TextArea
                          </legend>
                        </fieldset>
                        <div
                          class="mcm-input-unit-items css-1bcrtr8"
                        >
                          <input
                            autocomplete="off"
                            class="mcm-input-unit password-input css-1348qyx"
                            maxlength="30"
                            placeholder="Í¥ÄÎ¶¨Ïûê ÎπÑÎ∞ÄÎ≤àÌò∏Î•º ÏûÖÎ†•Ìï¥Ï£ºÏÑ∏Ïöî."
                            type="password"
                            value=""
                          />
                          <div
                            class="mcm-input-submit-button-wrapper css-cakho0"
                          >
                            <div
                              class="mcm-input-submit-button-items css-16nhp7q"
                            >
                              <button
                                class="mcm-close-button-unit css-12tnlcr"
                                type="button"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <button
                        class="mcm-button-unit css-1v4db1v"
                        id="admin-pw-toggle-btn"
                        role="button_click_event"
                        type="button"
                      >
                        <div
                          class="mcm-tooltip-wrapper css-s92xr2"
                        >
                          <div
                            class="mcm-tooltip-items css-rgebl6"
                          >
                            <div
                              class="mcm-tooltip-children css-uk6arh"
                            >
                              üôà
                            </div>
                          </div>
                        </div>
                      </button>
                    </div>
                  </div>
                  <button
                    class="mcm-button-unit css-83cl9t"
                    id="admin-login-btn"
                    role="button_click_event"
                    type="submit"
                  >
                    Î°úÍ∑∏Ïù∏
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </main>
</div>
`;
